lib/Logger.php

```diff
@@ -143,8 +154,17 @@ public function log($level, $message, array $context = array(), $muted = false)
         if (!defined('Productsup\Flexilog\Log\LogLevel::'.strtoupper($level))) {
             throw new \Psr\Log\InvalidArgumentException(sprintf('Level "%s" does not exist.', $level));
         }
-        foreach ($this->handlers as $handler) {
-            $handler->process($level, (string) $message, $context, $muted);
+
+        foreach ($this->handlers as $name => $handler) {
+            try {
+                $handler->process($level, (string) $message, $context, $muted);
+            } catch (Exception\HandlerConnectionException $e) {
```

Comment:
maybe even be less specific here. in any uncaught handler caused exception you would want the handler to be removed (or cause your HandlerException to be thrown)